volumetric/criticalKernelsThinning3D.cpp

```diff
@@ -253,11 +245,12 @@ int main(int argc, char* const argv[]){
    * Calculate distance map even if not requested:
    */
   trace.beginBlock("Create Distance Map");
-  using Predicate = Z3i::DigitalSet;
+  using Predicate = DGtal::functors::SimpleThresholdForegroundPredicate<Image>;
+  Predicate aPredicate(image, 0);
   using L3Metric = ExactPredicateLpSeparableMetric<Z3i::Space, 3>;
   using DT       = DistanceTransformation<Z3i::Space, Predicate, L3Metric>;
   L3Metric l3;
-  DT dt(obj.domain(),obj.pointSet(), l3);
+  DT dt(image.domain(), aPredicate, l3);
```

Comment:
Shouldn't the distance be calculated from the image set (that use custom min & max values provided by the user) instead of just the positive values of the image ?

Reply:
You are right.

Do you know the syntax to do that with the image_set in the DistanceTransformation?

I guess I could also create a predicate that uses the range of thresholds (instead of just thresholding values < 0)

Reply:
I may be wrong but I think you can directly pass the `image_set` as predicate like it was before: from what I understand, `Object::pointSet()` returns the `DigitalSet` given as parameter when constructing the `Object`, that is `image_set`. In addition, a `DigitalSet` is a `CPointPredicate` model.

Reply:
Fixed, thanks @rolanddenis